### ISSUE TITLE
How to launch a cluster using an instance in a private subnet with no internet access?

### ISSUE THREAD

**Issue by vbosquier** [aws/aws-parallelcluster#5336]
**Required Info:**
 - AWS ParallelCluster version 3.5.1
 - Please find the full cluster configuration + the command line used to create the cluster + the pcluster-cli.log contents in the file attached
 - Cluster name: test-private

**Bug description and how to reproduce:**
We are facing a problem that is very urgent to solve. We are running a HPC migration project to AWS for a very important customer in France. We need to launch a cluster with parallel Cluster in the customer's accounts. The network configured by the customer's teams will be **strictly private** with NO INTERNET ACCESS at all. The clusters will be launched using an instance directly set in the accounts with NO access to the internet either. We have tried to apply the following documentation (https://docs.aws.amazon.com/parallelcluster/latest/ug/network-configuration-v3.html#aws-parallelcluster-in-a-single-public-subnet-no-internet-v3) with an additional VPC ENDPOINT configured for the API Gateway service in order to use this endpoint to contact IAM (which sounded like a good idea at the beginning) with the following policy:

```
{
    "Statement": [
        {
            "Action": "execute-api:*",
            "Effect": "Allow",
            "Principal": "*",
            "Resource": "*"
        }
    ]
}
```

We need to find a way to run "pcluster create-cluster..." successfully in such network context where the instance running pcluster has no internet access. Note that the cluster does not even start, reason why I have no other log to attach than the pcluster-cli.log in the attached file.

Please can you help?
[test-api-gateway.txt](https://github.com/aws/aws-parallelcluster/files/11530059/test-api-gateway.txt)

Best regards,
Vincent.

**Comment by hanwen-pcluste** [comment]
Hi Vincent,

My apologies for the very late reply. https://docs.aws.amazon.com/parallelcluster/latest/ug/network-configuration-v3.html#aws-parallelcluster-in-a-single-public-subnet-no-internet-v3 is a good starting point. Was the cluster creation successful? If not, what is the error?

Thank you,
Hanwen

**Comment by cartalla** [comment]
Is there a reason why the node that creates the cluster has to be in the private VPC? Is there a reason they cannot use the UI? I've found the UI to be very helpful because you can configure and manage your cluster without needing a server or instance at all.

**Comment by vbosquier** [comment]
Hi.

@cartalla : The customer's security context drives a lot of constraints/rules among which:

- ALL the subnets in the AWS accounts in the project are supposedly NOT allowed to connect to any URL over the public Internet
- The PCluster clusters lifecycle management is considered as an "administration task", and this is the reason why it is mandatorily achieved using an administrator account on a machine that can only be reached with SSH through a bastion host

That's why we use the pcluster CLI in an EC2 instance.

@hanwen-pcluste : The cluster creation request fails before it actually start a cloudformation stack, because the URL to htps://iam.amazonaws.com is unreachable. Originally, I would have loved to have the possibility to rely on a VPC Endpoint for IAM to handle such situation... But it doesn't exist. That's why, as a fallback, I tried to use the VPC Endpoint of the API Gateway Service (workaround solution that remains unsuccessful unfortunlately until now...).

Let me know what you think.

Best regards,
Vincent.

**Comment by lukeseawalker** [comment]
Hello,
I believe the error is coming from a CLI validator, not being able to contact IAM service to perform some param validation.

As a workaround I would suggest you to disable the validator in question, see https://docs.aws.amazon.com/parallelcluster/latest/ug/pcluster.create-cluster-v3.html
```
pcluster create-cluster [...] --suppress-validators type:<TheFailingValidatorName>
```
where `<TheFailingValidatorName>` is the validator failing the check.

**Comment by vbosquier** [comment]
OK thanks.
We have finally used this workaround successfully with Kevin TUIL in a fully private environment  with
--suppress-validators type:AdditionalIamPolicyValidator

And it worked.
We also found out how install PCluster script from your standalone installation procedure to install PC v3.6.0 on a Management Host which does not have an internet connection (no pip install allowed). I think we're OK for that...
Thanks a lot.
